File: frontend/src/app/shared/service/admin/course/ParcoursAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {ParcoursDto} from 'src/app/shared/model/course/Parcours.model';
import {ParcoursCriteria} from 'src/app/shared/criteria/course/ParcoursCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class ParcoursAdminService extends AbstractService<ParcoursDto, ParcoursCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): ParcoursDto {
        return new ParcoursDto();
    }

    public constrcutCriteria(): ParcoursCriteria {
        return new ParcoursCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/parcours/';
    }
}

File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/course/ParcoursAdminService.java
package ma.zs.zyn.service.facade.admin.course;

import java.util.List;
import ma.zs.zyn.bean.core.course.Parcours;
import ma.zs.zyn.dao.criteria.core.course.ParcoursCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface ParcoursAdminService extends  IService<Parcours,ParcoursCriteria>  {

    List<Parcours> findByCentreId(Long id);
    int deleteByCentreId(Long id);
    long countByCentreRef(String ref);




}
